### PR TITLE
Enable profiling over HTTP in debug mode

### PR DESCRIPTION
**Pull request by aluzzardi**
Signed-off-by: Andrea Luzzardi <aluzzardi@gmail.com>

### CODE REVIEW

**Comment by abronan** [comment]
LGTM

**Comment by dongluochen** [comment]
LGTM.

**Comment by MHBauer** [comment]
Cool! I've never seen this before.

Why no `Trace`?

**Comment by aluzzardi** [comment]
@MHBauer What do you mean?

**Comment by MHBauer** [comment]
@aluzzardi  maybe I am looking in the wrong place, but I see one more function called Trace, [here](https://golang.org/src/net/http/pprof/pprof.go?s=6283:6333#L72)

File: api/primary.go [hunk 1]
```diff
@@ -100,8 +102,20 @@ func writeCorsHeaders(w http.ResponseWriter, r *http.Request) {
...
+	var r = mainRouter.PathPrefix(path).Subrouter()
+	r.HandleFunc("/pprof/", pprof.Index)
+	r.HandleFunc("/pprof/cmdline", pprof.Cmdline)
+	r.HandleFunc("/pprof/profile", pprof.Profile)
+	r.HandleFunc("/pprof/symbol", pprof.Symbol)
+	r.HandleFunc("/pprof/block", pprof.Handler("block").ServeHTTP)
+	r.HandleFunc("/pprof/heap", pprof.Handler("heap").ServeHTTP)
+	r.HandleFunc("/pprof/goroutine", pprof.Handler("goroutine").ServeHTTP)
+	r.HandleFunc("/pprof/threadcreate", pprof.Handler("threadcreate").ServeHTTP)
+}
```
Comment (vieux) [inline api/primary.go]: `r.HandleFunc("/debug/pprof/trace", pprof.Trace)`

Comment (MHBauer) [inline api/primary.go]: Yes, exactly this. I probably should have done this by example like you did. If there is a reason to not expose it, that's fine. Don't let me hold up the merge.

Comment (vieux) [inline api/primary.go]: It's new in go 1.5.x, I added a commit to the PR

**Comment by vieux** [comment]
LGTM

**Pull request merged**